gradle/java9plus.gradle

```diff
@@ -2,42 +2,26 @@ def javaInstallationForTest = rootProject.availableJavaInstallations.javaInstall
 def currentJavaVersion = javaInstallationForTest.javaVersion
 
 if (isSupportedJavaVersion(currentJavaVersion)) {
-
     tasks.withType(org.gradle.testing.IntegrationTest) {
```

Comment:
do you have any insight on when the test should stay here vs. using the `JDK8_OR_EARLIER` or `FIX_TO_WORK_ON_JAVA9` `TestPrecondition`?

Reply:
Optimally, we'd fix all tests and wouldn't have to do either. I think it will simply not happen as some of the external tools do not support Java 9 yet (e.g. Play) or will never support it (e.g. FindBugs).

There's are benefits to both approaches.

1. Having the exclusion in the build script provides a central place but sometimes is too coarse-
 grained (sometimes only specific test cases are failing).
2. Using `TestPrecondition` in the test class makes them harder to find.

We should probably decide on a case by case basis.

Reply:
I would think that having both (this file and the `TestPrecondition`) makes them harder to find.  If they were all marked with a `TestPrecondition`, it would be a simple "Find Usages".  Analogously, if they could only appear in this file, then it would be entirely centralized.

It also seems like using this file for tests that will never work with Java 9 (or 10 or 11 or etc.) reduces the signal-to-noise ratio.  Taking FindBugs as an example of a library that won't ever support Java 9; if the associated test classes used `TestPrecondition.JDK8_OR_EARLIER`, it's clear that no additional cycles need to be devoted to those tests when the next batch of Java 9 (or 10 or 11) test exclusions are worked on.

Reply:
I agree. My gut feeling says that we should rather use the `TestPrecondition` notation.

Reply:
+1 for moving towards annotations. Those also work in the IDE